Wasedasai2022

SnakeSneakS opened the pull request:
[これ](https://github.com/MISW/event-misw-jp/pull/17) の継承

# プルリクエストを行う前に、以下のことを確認してください。

- [ ] コードのフォーマットを行いましたか？
- [ ] エラーや警告を出さずに、正しくビルドされることを確認しましたか？
- [ ] 様々なブラウザ・OS・解像度でテストを行い、すべて正常に表示されましたか？
- [x] サイズが 25 MiB を超えるファイルが存在しませんか？
- [x] ライセンス感染するパッケージやライブラリ等を使用していませんか？ (例: GNU GPL・CC BY-SA )
- [ ] サインオフを行いましたか？ (例: Signed-off-by: mischan <miswchan@misw.jp> )
- [x] プルリクエストを送信したら、Web の担当者と SysAd の担当者にコードレビューをリクエストしてください。
- [x] 新しい特設サイトを追加する場合は、[misw-jp](https://github.com/MISW/misw-jp) のバナーも更新してください。

# プルリクエストの内容を以下に記載してください。
早稲田祭2022特設サイトの設置
# 特筆すべき内容があれば以下に記載してください。
・iPhoneでフォントが変わってしまう問題があります
・iPhoneで背景画像のみすちゃんが表示されない問題があります

kkosugitm committed:
Wasedasai2022 (#17)

CaffeeLake reviewed (changes requested):
https://github.com/MISW/event-misw-jp/pull/17#pullrequestreview-1161073766
https://github.com/MISW/event-misw-jp/pull/17#issuecomment-1296153575
https://github.com/MISW/event-misw-jp/pull/17#issuecomment-1296154520

CaffeeLake committed:
CI: Pass

kkosugitm committed:
著作権表記を変更

CaffeeLake commented:
- [x] 文字が見にくい。
![image](https://user-images.githubusercontent.com/93961902/198871132-b8837c94-51c7-425a-81e0-25e1e722126b.png)

- [x] 下に謎のスペースがある。
![image](https://user-images.githubusercontent.com/93961902/198871058-0e8973a6-5b20-40bc-8035-c75f1a93936a.png)

- [x] 見にくい。
![image](https://user-images.githubusercontent.com/93961902/198871304-3e332e99-7196-469f-95a2-a51100328f74.png)

kkosugitm committed:
みすちゃんの背景画像を軽量化

kkosugitm committed:
PC版でのAboutの文字を読みやすく改善

kkosugitm committed:
フッターのレイアウトを変更

kkosugitm committed:
埋め込む動画を変更

kkosugitm committed:
複数の問題を修正

・Contentsに同人誌企画を追加
・一部の環境で背景のみすちゃんが隠れる問題を修正
・Googleフォントを導入し、環境によってフォントが変わる問題を修正

kkosugitm committed:
色やレイアウトを見易いように調整

kkosugitm committed:
スマホ版のみすちゃんの表示を調整

kkosugitm committed:
背景のレイアウトを調整

kkosugitm committed:
ホームの表示崩れを修正

cherr0406 reviewed (changes requested):
お疲れ様です。ほとんど完成ですね、ありがとうございます。
以下の点について見直していただきたいです。

## コードに関して

- ヘッダー部分にOGP画像（TwitterなどのSNSに投稿する際に見える画像）と、それを指定するコードを付け足してほしいです。過去の早稲田祭サイト等を参考にしてみてください。metaタグの`property="og:xxx"`, `name="twitter:xxx"`のように書かれているところを足して、書き換えてもらえればOKです。
- OGP画像は1200*630のサイズで、真ん中に重要な情報が来るようにするとよいです。[このサイト](https://ogimage.tsmallfield.com/)を使ってシミュレーションすると便利です。
- Twitter、LINE、Facebook共有ボタンの追加をお願いします。こちらも過去の早稲田祭サイトを参考にするとよいです。場所はフッターが一番適切かと思います。


## 見た目に関して
【PC表示】
- トップ左下の動画ですが、もう少し大きくても良いかもしれません。余裕があれば、クリックすると大きくモーダルウィンドウが出てきて、そこで再生できるようにすると良いと思います。
- Contentsの作品ですが、画像やタイトル文字列クリックで作品に飛べるようにする予定はありますか？　もしあればこのままでいいのですが、ない場合は「作品はすべて教室展示です」等の文言を入れてあげると親切です。初見ではリンク忘れかな？と思ってしまいそうです
- Accessの右半分の文章、左右にもう少しpaddingあったほうよさそうです
- フッターの製作者表記とコピーライト表記が詰まって見えているので、このまま縦並びにするのであれば余白を取るか、横線を入れてあげるといい気がします。もしくは、横並びにしてもいいと思います。

【スマホ表示】
- Accessの教室案内が変なところで改行されがちなので、3号館～の前で改行を挟んであげると見栄えがよさそうです。
- PC表示同様、フッターの製作者表示とコピーライトの間にもう少し余白があるといいかもしれません。

kkosugitm committed:
OGP画像の追加

kkosugitm committed:
複数の修正

・OGP画像を設定
・フッターにSNSシェアボタンを設置
・変なところで改行が入らないように調整
・Aboutとフッターの要素間の余白を調整

kkosugitm committed:
複数の修正

・Contentsにノベルゲーム企画を追加
・共有ボタンのリンク先を修正
・ホームのレイアウトを調整
・同人誌閲覧機能を追加
・ボタンホバー時のアニメーションを追加

kkosugitm committed:
Update style.css

CaffeeLake committed:
CI: Pass

kkosugitm committed:
OGPと共有ボタンの微修正

kkosugitm committed:
metaタグのOGP設定、共有ボタンの微修正

kkosugitm commented:
お疲れ様です。iPhoneの問題以外はおおむね対応完了しました。
Contentsの作品については、要望があった同人誌企画は閲覧ページへのリンクを作りました。ボタンの有無でリンクの有無も判断できるかな？と思うのですが、いかがでしょうか？やはり「作品は教室展示です」の文言が必要であれば追加します。

kkosugitm committed:
スマホ表示時にナビゲーションボタンがiPhoneのナビゲーションバーに干渉しないように調整

kkosugitm committed:
直前のpushを元に戻しました（safariでは干渉しないと判明したため）

kkosugitm committed:
微修正

kkosugitm committed:
レイアウト、色を全体的に調整

kkosugitm committed:
Update style.css

kkosugitm committed:
Update style.css

kkosugitm committed:
Update style.css

kkosugitm committed:
Update style.css

CaffeeLake commented:
MISNOTE5、PDFファイルか画像ファイルかどちらかでいいのでは。
ブラウザでPDFを表示させることはできるし、画像が見えるならPDFはいらないと思うし。

CaffeeLake committed:
CI: Pass

kkosugitm commented:
AndroidだとブラウザでPDFを表示できないんですよね...（少なくともGoogleアプリとChromeとFireFoxはできないはず）
都度ダウンロードしてPDFビューワーアプリ開かなくちゃいけないので、気軽に読める画像ときれいなPDF両方用意しました。
どちらか消すとしたらPDFなくして画像だけに、ですかね...？

CaffeeLake commented:
あれ、私Androidを使っていますが、ブラウザ上でPDFを表示できますよ。

kkosugitm commented:
ありゃ、ほんとですか
何のブラウザ使ってますか？

CaffeeLake commented:
使ってるjsこれっぽい。
https://github.com/mozilla/pdf.js

CaffeeLake commented:
ブラウザはChromeとFirefoxどっちでも動きました。

CaffeeLake commented:
これAndroidでも表示されるはず。
https://mozilla.github.io/pdf.js/web/viewer.html

kkosugitm commented:
どうやらAndroid 7.0以降から表示できなくなったみたいですね。。困った仕様です。
pdf.jsは表示はできるんですが色がおかしくなっちゃいます。言葉だと難しいのでdiscordに動画送ります。

CaffeeLake commented:
Android 12 だけど、私の端末ではちゃんと表示されてるんですよね...。

kkosugitm commented:
https://photos.app.goo.gl/L9uu5xS6aCKi3Pvs7
↑ discordで送れなかったので
なんででしょうね。。。

CaffeeLake commented:
あと、PDFファイル、プロパティに作成者の本名が載っているので、不特定多数が見るサイトなので、本名の部分は消したほうがいいと思います。
![5](https://user-images.githubusercontent.com/93961902/199148585-cc129ffd-f819-4506-9a76-9b7184c9e3e2.jpg)

kkosugitm commented:
あ、ごめんなさい。差し替えます

kkosugitm committed:
MISNOTE5のPDFファイルを差し替え

kkosugitm committed:
Update style.css

kkosugitm committed:
Update style.css

cherr0406 commented:
お疲れ様です。コメントした部分も修正ありがとうございます。

> やはり「作品は教室展示です」の文言が必要であれば追加します。

「読んでみる」ボタンのおかげで差別化できたので、追加の必要はない気がします。

動画の大きさに関しては、今のままでも視聴に問題はないので、そのままでもOKです。（気が向いたらなにかコミットするかも）

> AndroidだとブラウザでPDFを表示できない

昨年度はPDFを期間限定で公開しようとしていて、DL不可にするためGoogle Driveに飛ばしていました。今年度はDL可能にするのかな？と思いますがそのあたり一応確認したほうが良いかもしれません

kkosugitm committed:
企画を追加

kkosugitm commented:
企画長のNOAHさんに確認しました。早稲田祭後もそのまま公開で問題ないそうです

cherr0406 commented:
了解です。ではサイト上で公開でよさそうですね。

画像とPDFどちらも含めるのは重くなるので、PDFのみがいいと思います。他サイトでも通常はPDFを開くことができる前提で構築されているので、「正常に表示できない場合はPDFをダウンロードしてご覧ください」等の文言を入れて、画像は含めない方向で行きたいです。あと画像で載せてしまうと、それぞれ単体で画像検索に引っ掛かりそうですし、個人的にあんまり好みではないです…

あと、見た目に関してもう一点だけ確認してもらいたいのが、トップ左側にある円(#left-circle)が画面サイズと解像度によって歪んでしまうようです。おそらく縦並びFlexにしている関係だと思われますので、正円になるよう固定してもらえるといいかなと思います。右側の円(#date-circle)も同じ仕様になっていると思うので、両方修正してほしいです。

それ以外については、Chrome/Edge/FirefoxのPC表示とSafariでのスマホ表示を確認し、サイトの表示崩れ等の問題はありませんでした。Androidは所持していないのでCaffeeくんに確認お願いしたいです。

kkosugitm committed:
複数の修正

・iPhone, iPadでの背景表示の問題を解決（べびさんありがとうございました）
・ウルトラワイドモニターなどの解像度でホームの円形要素が歪まないように調整

kkosugitm committed:
コードをフォーマット

kkosugitm commented:
残りの修正点は後程コミットします。
直前のコミットでスマホ版サイトの円形の場所がおかしくなってますが、こちらも後で修正します。

kkosugitm committed:
複数の修正

・スマホ版のホームの表示崩れを修正
・doujinshi.html を廃止
・ペンギン企画のアプリストアバッジを追加
・Aboutのレイアウトを調整

kkosugitm commented:
お待たせしました。すべて修正完了しました！

kkosugitm committed:
不要な画像を削除

kkosugitm committed:
Update index.html

CaffeeLake commented:
PDFに関してですが、いちいちダウンロードして開くのは手間だと思うので、前提としてWebサイト内でPDFが見れる状態にしておいて、もし表示できないならダウンロードしてみてください、っていう感じでお願いします。

CaffeeLake commented:
UIに関しては、2点気になることがあります。
スマホで表示した際に、左下のAboutを押すと、MIS.Wのロゴとみすちゃんの顔が被るので、被らないように調節してほしいです。
早稲田祭の中の出典なので、キャンパスの場所よりも、キャンパス内のでの位置が求められていると思うので、地図のズームレベルは、最初からもう少し拡大してもいいと思います。

kkosugitm commented:
pdf.jsを導入するっていう認識で合ってますか？

CaffeeLake commented:
PDFを埋め込む方法はいくつもあるので、難しければ他の方法でも大丈夫です。

CaffeeLake commented:
最悪、コード内のライセンス表記を削除しなければ、公式サンプルからPDFの中身を差し替えたやつでも大丈夫だと思います。
表記を消したらアウトですが。

kkosugitm committed:
pdf.jsを導入、Googleマップの縮尺を変更

kkosugitm committed:
Update style.css

kkosugitm commented:
これでどうでしょう
Googleマップの縮尺はあんまり微調整が効かなくて、もう一段階拡大するとめちゃでかくなっちゃいます。

kkosugitm commented:
手持ちのGoogle, Samsung, Xiaomi, HUAWEI端末で確認したところ、Google端末でのみPDFがおかしくなるからPixel固有の問題かも...？

CaffeeLake commented:
Firefox(Windows)でも表示されない気がする。
![image](https://user-images.githubusercontent.com/93961902/199476366-f32acc5e-a009-4868-b5fd-f34273c7044f.png)

kkosugitm committed:
Update index.html

kkosugitm committed:
Update index.html

kkosugitm commented:
こちらの環境ではFireFoxでも表示できるなあ...
![image](https://user-images.githubusercontent.com/55230829/199477523-08e6e6f6-77e5-46da-9a6d-8fec0687a89e.png)

CaffeeLake committed:
Fix: PDF.js

CaffeeLake committed:
Delete MISNOTE5.pdf

CaffeeLake commented:
これによってBuildフォルダーがアップロードされていない。

https://github.com/MISW/event-misw-jp/blob/a5b120a4687a781aa463ba13aa217ece2b75a046/.gitignore#L20

CaffeeLake committed:
Fix: Ignore pdfjs/build

CaffeeLake commented:
.gitignore にマッチするファイルは、手元では存在していても、commitされないので、手元では動いても、デプロイ先では動かないようなことが存在する。
これは、一時的に保存したいものや、動かすのに必要だけど、GitHubで公開されたくないキーファイルとかに利用される。

CaffeeLake commented:
あと2つ気になった点として、
TopとAboutとFooterの背景色は、右端から左端まであるのに対し、ContentsとAccessの背景色は右端と左端に白い空間があるので、統一されていなく少し違和感がある。
CG研画集の後ろだけ背景色がない。

kkosugitm commented:
なるほどそういうことか。。

TOP、ヘッダー、フッターはwrapperをかけないっていうのは割とよく使われる見せ方な気が。
まあAboutは微妙なところだけど、、みすちゃんを文字と被らせず広く見せるためにそうしてる

CG研画集のところはちょっと考えます。

kkosugitm commented:
TOPは白背景でもいいかもしれない
![image](https://user-images.githubusercontent.com/55230829/199490057-6f44d489-9835-4278-bbe2-90b06cd0d30a.png)

CaffeeLake commented:
ヘッダーとフッターに左右の空白を設けるのではなく、その逆で、ヘッダーとフッターの背景色はいいと思うのですが、WQHDや4K解像度のモニターで表示すると、ContentsやAccessが以下のような感じになるんですよね。
あとフッターの文字も広がりすぎるのも気になるけど。

![image](https://user-images.githubusercontent.com/93961902/199495097-7ed5ec47-be5f-4815-876d-7080b384c7ca.png)

kkosugitm committed:
レイアウト調整

kkosugitm committed:
Merge branch 'wasedasai2022' of https://github.com/MISW/event-misw-jp into wasedasai2022

kkosugitm commented:
コンフリクトしてCafeeLakeさんの直前のコミット破壊しちゃったかもしれません
同人誌企画の読んでみるボタン押したら機能しなくなってたので
ほんとごめんなさい。。

kkosugitm commented:
wrapperのwidthは少し拡大しておきました。
なるほど、自分も4Kモニタ使ってますが100%表示にしたことないので気づきませんでした
WQHDユーザーで100%で使ってる人どれくらいいるんですかね...

とりあえず一度拡大率下げてみて考えます

kkosugitm committed:
不明

kkosugitm commented:
うーん、WindowsはWQHDの場合デフォルトの拡大率が200%に設定されているようなので、さほど問題ない気がするんですよねえ

CaffeeLake commented:
コンフリクトして破壊するのは、最初のうちは誰でもやることなので、全然気にしなくて大丈夫です。

CaffeeLake committed:
Revert: Conflict

CaffeeLake committed:
Revert: Conflict

kkosugitm committed:
Update style.css

kkosugitm committed:
Merge branch 'wasedasai2022' of https://github.com/MISW/event-misw-jp into wasedasai2022

kkosugitm commented:
>あと拡大したら高解像度のモニターを使う利点がないので、普通は100%に設定すると思います。

そうですかね。。文字やら画像やらが見やすくなるという利点がありますし。100％だと文字が小さくて自分は読めないので拡大してます。そのために推奨設定が用意されてるんでしょうし。。
画面の大きさでデフォルトが変わるのは初耳でした。
TOPとフッター以外の青背景の横幅を統一したのと、高解像度でも端の方まで背景が来るようにしました。

kkosugitm committed:
背景幅調整

CaffeeLake commented:
そもそも小さい文字が読みにくい人間なら、小さい画面に高解像度を詰め込む必要あります?
4KでもFHDでも変わらないんじゃ。
少なくとも、クリエーターやプログラマープログラマーの人は、一度に表示できる情報量が求められているので、拡大はしない傾向にあるとおもいます。
4K解像度をFHDの作業領域に制限して200%に拡大するくらいなら、最初からFHDでいいじゃんと私は思いますね。
まぁ動画とかしか見ない人間なら、どっちでも問題ないかもですね。

CaffeeLake reviewed (approved):
lgtm

kkosugitm commented:
100%のFHDと拡大した4Kでは相当見やすさ違うと思いますよ（こればっかりは主観でしかないので論点にしても仕方ない感あります...）。
Cafeeさんのおっしゃる通り、プログラマや一部のクリエイターが表示領域の拡大のために使ってるのなら拡大は本末転倒ですね。
ゲームする人、動画見る人、↑とは別種のクリエイター（写真目的など）は拡大して使っても十分に意味はあるんじゃないでしょうか。ゲームならより没入感のある映像が見られますし、フォトショなどで画像加工する際も細部の情報量が違います。
そもそも意図して高解像度モニタを買ってない（Macやサーフェスなど選ぶまでもなく付いてくる）ケースもあるかと。
要するに、100%で使うメリットを否定するわけではありませんが、みんながみんな上記の「プログラマや一部のクリエイター」ではないわけで。Cafeeさんが示した13.42%の中でもそういった人はさらに限られてくるんじゃないかなー（これも主観です）と思った次第です。しかもこの統計はSteamのものなので、必然的にゲームユーザーの回答が多くなります。通常よりも高解像度モニタが求められやすい母集団からの統計ですから、13.42%という数字自体どこまで参考になるのかも疑問です。

CaffeeLake commented:
ゲームや画像編集ソフトの綺麗さは、そもそも拡大率は関係ない気がします。
拡大しても等倍でも物理的な解像度は変わらないので。
細かい部分を編集したいなら、画像を拡大すればいいだけなので。
Steamの統計なので、おっしゃる通り、実際とは偏っている可能性はあります。
ゲームをする人は高解像度を求める人もいますが、それと同じくらい高フレームレートを求める人がいると思うので、バランスがとれていると私は思います。
意図して高解像度モニタを買ってないのは、一定数いると思うし、文字が大きいほうがいい人間もいるのはわかります。
確かに、ノートPCとかに4K解像度を積んでいた場合、拡大しないと文字を読むのが難しいと思います。
でも27インチとかですと、100%でもそんなに小さくはありません。
いろいろ話はそれましたが、私が言いたいのは、4KやWQHDモニターで100%で利用している人も一定数はいるということです。
あなたはデフォルトが200%表記だからさほど問題ないと言っていても、そのデフォルトは画面サイズによって異なるし、等倍で使いたい人間もいるので、問題が発生する環境が一定数はあるということです。
その一定数がどれくらいいるかは、統計がないのでわかりませんが。

kkosugitm commented:
>ゲームや画像編集ソフトの綺麗さは、そもそも拡大率は関係ない気がします。

そうです、関係ありません。「あと拡大したら高解像度のモニターを使う利点がない」「4K解像度をFHDの作業領域に制限して200%に拡大するくらいなら、最初からFHDでいいじゃんと私は思いますね。」に対する回答として書きました。

>4KやWQHDモニターで100%で利用している人も一定数はいるということです。

それは私も認めるところです。13.42%という数字を出されたことから、対応しないといけない理由が「少数のケースなら労力割いて対応する必要はないが、そうではないから」だと解釈しました。今の返信を見る限り「そういう人がいないわけではないから」という理由ですかね。
「その一定数がどれくらいいるかは、統計がないのでわかりませんが。」とおっしゃっている通り、多い少ないの議論はこれ以上しても結論が出ないと思うのでおしまいにします。

>問題が発生する環境が一定数はあるということです。

とりあえず昨日の修正で問題ないですか？

CaffeeLake commented:
拡大する利点がないというのは、確かに言いすぎました。
「少数のケースなら労力割いて対応する必要はない」という最初の解釈の前半の部分のほうで間違っていないです。
4Kみたいな少数派に合わせる必要はないけど、WQHDはそこそこいるから対応したほうがいいよね、という話です。
そもそも高解像度のモニターは画面サイズが大きいものが多いので、デフォルト設定のままで使っていても100%のデバイスは多いと思います。
「そういう人がいないわけではないから」という意味で言ったわけではなく、「そういう人が少数ではなくそこそこいる可能性があるから」という意図で言いました。
拡大率に関する意見は、少し私の考えを当たり前のように述べていてしまった点があるので、それは申し訳ないです。

Webページについては、昨日の変更で問題ないです。
コードレビューで承認しています。

細かいところを色々と指摘してしまいましたが、きちんと対応して頂てありがとうございました。
私はこれでデプロイしても問題ないと思います。
他の人からまだ指摘が入るかもしれませんが、私のほうからは取り合えず、お疲れさまでした。
途中で投げ出さすによく頑張ったと思います。

kkosugitm commented:
いえいえ、手伝って頂いてありがとうございました。分からない部分も多いので助かりました。
了解です。ドット絵ゲーム企画の画像をもう少し大きく切り抜いてほしいと言われたので、この後コミットしますね。
サイト公開後に修正する場合（CG研画集の追加など）はこれまで通りコミットして問題ありませんか？

SnakeSneakS commented:
いや、プルリク出して一旦他の人の目を通して欲しい

CaffeeLake commented:
> いや、プルリク出して一旦他の人の目を通して欲しい

とのことなので、今まで通りPR出してください。

kkosugitm committed:
Update dottoe.png

kkosugitm commented:
了解です

SnakeSneakS commented:
あとはまいなさんかしら、よさそう..?
@maina82c8

CaffeeLake commented:
早稲田祭係に対してはDiscordでメンションしておいた。

CaffeeLake commented:
早稲田祭係からは
> とてもよいです！ありがとうございます

だそうです。

cherr0406 reviewed (approved):
はい、特に問題ありません。修正をお願いした個所は直してもらっていますね。
お疲れさまでした～
デプロイ後、OGPが機能しているかだけ確認したら、全体告知してOKです。早稲田祭係の方にもお伝えください。

CaffeeLake commented:
あとはバナー。

cherr0406 commented:
バナーはやっておきます👌

CaffeeLake commented:
あ、コメント付く前にやってしまった。

cherr0406 commented:
助かります、バナーもOGPも大丈夫そうですね。ひとまず完了です。お疲れ様でした！
